model-optimizer/extensions/front/tf/RollRealImagPackReplacement.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from extensions.middle.InsertLayoutPropagationTransposes import mark_input_as_in_correct_layout
+from extensions.ops.roll import Roll
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.common.replacement import FrontReplacementSubgraph
+from mo.front.subgraph_matcher import SubgraphMatch
+from mo.graph.graph import Graph, Node, rename_nodes
+
+
+class RollRealImagPackReplacement(FrontReplacementSubgraph):
+    """
+    Some TF models contain Roll for complex data, as a part of the sub-graph
+
+        input   shift   axes
+          |       |      |
+         -------------------
+                 Roll
+                  |
+         -------------------
+         |                 |
+        Real              Imag
+         |                 |
+         -------     -------
+               |    |
+                Pack
+                  |
+                SomeOp
+
+    This sub-graph can be replaced with the sub-graph
+
+        input   shift   axes
+          |       |      |
+         -------------------
+                 Roll
+                  |
+                SomeOp
+
+    But after such replacement, we should correct axes of Roll, because input data are real now. Namely, if
+    there are negative axes for Roll, we need subtract 1 from such axes indices.
+
+    The transformation RollRealImagPackReplacement performs the above mentioned replacement.
+    """
+    enabled = True
+
+    def run_after(self):
+        from extensions.front.tf.SSliceComplexRollReplacement import SSliceComplexRollReplacement
+        return [SSliceComplexRollReplacement]
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('unroll', dict(op='Roll')),
+                ('real', dict(op='Real')),
+                ('imag', dict(op='Imag')),
+                ('pack', dict(op='Pack')),
```

Comment:
This transformation should be executed before running transformation model-optimizer/extensions/front/Pack.py which replaces operations of type "Pack"

Reply:
Done.